pages/web/web-paas/development-variables/guide.en-ca.md

```diff
@@ -5,253 +5,268 @@ section: Development
+| PLATFORM_PROJECT          | Yes   | Yes     | The project ID |
+| PLATFORM_TREE_ID          | Yes   | Yes     | The ID of the tree the application was built from, essentially the SHA hash of the tree in Git. Use when you need a unique ID for each build |
+| PLATFORM_PROJECT_ENTROPY  | Yes   | Yes     | A random, 56-character value created when the project is created and then stable throughout the project's life. Can be used for Drupal hash salts, Symfony secrets, and other similar values. |
+| PLATFORM_APP_DIR          | Yes   | Yes     | The absolute path to the application directory. |
+| PLATFORM_SOURCE_DIR       | No    | No      | Equivalent to `PLATFORM_APP_DIR` in the context of a running [source operation](../configuration/app/source-operations.md). The directory contains a writable copy of your repository that you can commit to during the operation. |
+| PLATFORM_APPLICATION_NAME | Yes   | Yes     | The application name as set in the `.platform.app.yaml` file. |
+| PLATFORM_APPLICATION      | Yes   | Yes     | A base64-encoded JSON object that describes the application. It maps certain attributes from your `.platform.app.yaml` file, some with more structure. See [notes](#platform_application). |
+| PLATFORM_BRANCH           | No    | Yes     | The name of the Git branch. |
+| PLATFORM_DOCUMENT_ROOT    | No    | Yes     | The absolute path to the web document root, if applicable. |
+| PLATFORM_ENVIRONMENT      | No    | Yes     | The name of the Web PaaS environment. |
+| PLATFORM_ENVIRONMENT_TYPE | No    | Yes     | The type of the Web PaaS environment (development, staging or production). |
+| PLATFORM_SMTP_HOST        | No    | Yes     | The SMTP host to send email messages through. Is empty when mail is disabled for the current environment. |
+| PLATFORM_RELATIONSHIPS    | No    | Yes     | A base64-encoded JSON object of relationships. The keys are the relationship name and the values are arrays of relationship endpoint definitions. The exact format is defined for each [service](../configuration-services). |
+| PLATFORM_ROUTES           | No    | Yes     | A base64-encoded JSON object that describes the routes for the environment. It maps the content of the `.platform/routes.yaml` file. |
 
+Dedicated instances also have the following variables available:
 
-Since values can change over time, the best thing is to inspect the variable at runtime then use it to configure your application. For example:
+| Variable name    | Build | Runtime | Description |
+| ---------------- | ----- | ------- | ----------- |
+| PLATFORM_MODE    | No    | Yes     | `enterprise` in all production and staging Dedicated environments. Note that an Enterprise support plan doesn't always imply a Dedicated environment, but a Dedicated environment always implies an Enterprise support plan. |
```

Comment:
External service labels / explanation

Reply:
These variables are within Platform.sh environment, i guess explanation should not be an issue.